ext/cluster/cluster.go

```diff
@@ -0,0 +1,186 @@
+package cluster
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	v2 "github.com/kyverno/kyverno/api/kyverno/v2"
```

Comment:
not sure if it will work this way. It basically creates a circular dependency when this package depends on kyverno itself. Thats why I said that this packages would need to move as well or you need to add an abstraction for it.